### PR TITLE
2.1 Sample updates (Round 2)

### PR DESCRIPTION
**Pull request by guardrex**
Addresses #5495 

Takes care of topics and samples in the *fundamentals/configuration* TOC area ...

* fundamentals\url-rewriting.md (1 sample)
* fundamentals\configuration\options.md (1 sample)
* fundamentals\configuration\platform-specific-configuration.md (2 samples)

### CODE REVIEW

**Comment by guardrex** [comment]
@Tratcher Ran into a problem on one of these and need a tip.

This worked before ...

```csharp
.UseKestrel(options =>
{
    options.Listen(IPAddress.Loopback, 443, listenOptions =>
    {
        listenOptions.UseHttps("testCert.pfx", "testPassword");
    });
});
```
... but it's throwing now ...

> An attempt was made to access a socket in a way forbidden by its access permissions

I have the option of dropping this (`AddRedirectToHttpsPermanent`) example from the sample, but I'd prefer to get it working if possible.

**Comment by Tratcher** [comment]
That error means some other server is still using port 443, there's likely nothing wrong with the sample. Make sure you kill other server instances and try again.

File: aspnetcore/fundamentals/configuration/platform-specific-configuration/sample/StartupDiagnostics/StartupDiagnostics.csproj [hunk 1]
```diff
@@ -1,44 +1,44 @@
...
+    <PackageReference Include="Microsoft.AspNetCore.Hosting" Version="2.1.0" />
+    <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="2.1.0" />
+    <PackageReference Include="Microsoft.AspNetCore.Hosting.Abstractions" Version="2.1.0" />
   </ItemGroup>
 
   <Target Name="CopyTarget" AfterTargets="Build">
     <Message Importance="High" Text="Creating directory for assembly if it doesn't exist ..." />
-    <MakeDir Directories="$(USERPROFILE)\.dotnet\store\x64\netcoreapp2.0\StartupDiagnostics\1.0.0\lib\netcoreapp2.0\" 
-             Condition="!Exists('$(USERPROFILE)\.dotnet\store\x64\netcoreapp2.0\StartupDiagnostics\1.0.0\lib\netcoreapp2.0\')" />
+    <MakeDir Directories="$(USERPROFILE)\.dotnet\store\x64\netcoreapp2.1\StartupDiagnostics\1.0.0\lib\netcoreapp2.1\"
```
Comment (scottaddie) [inline aspnetcore/fundamentals/configuration/platform-specific-configuration/sample/StartupDiagnostics/StartupDiagnostics.csproj]: Consider replacing occurrences of "netcoreapp2.1" with `$(TargetFramework)` to improve maintainability. When the TFM changes above, you don't have to update all these locations too.

File: aspnetcore/fundamentals/configuration/platform-specific-configuration.md [hunk 2]
```diff
@@ -63,7 +63,7 @@ Only part of the file is shown. The assembly name in the example is `StartupEnha
 
 ### Update the dependencies file
 
-The runtime location is specified in the *\*.deps.json* file. To active the enhancement, the `runtime` element must specify the location of the enhancement's runtime assembly. Prefix the `runtime` location with `lib/netcoreapp2.0/`:
+The runtime location is specified in the *\*.deps.json* file. To active the enhancement, the `runtime` element must specify the location of the enhancement's runtime assembly. Prefix the `runtime` location with `lib/netcoreapp2.1/`:
```
Comment (scottaddie) [inline aspnetcore/fundamentals/configuration/platform-specific-configuration.md]: I suggest making this more generic by replacing `netcoreapp2.1` with a placeholder like `<target_framework_moniker>`.

File: aspnetcore/fundamentals/configuration/platform-specific-configuration/sample/StartupDiagnostics/StartupDiagnostics.csproj (hunk 1 above)
Comment (guardrex) [reply aspnetcore/fundamentals/configuration/platform-specific-configuration/sample/StartupDiagnostics/StartupDiagnostics.csproj]: @scottaddie Great idea. How about we continue with ...

```xml
<SharedFrameworkVersion>2.1.0</SharedFrameworkVersion>
<StartupDiagnosticsVersion>1.0.0</StartupDiagnosticsVersion>
```
... and ... `$(SharedFrameworkVersion)` + `$(StartupDiagnosticsVersion)`

**Pull request merged**